### DISCUSSION TITLE
High Host CPU use. Cubase 11-2x windows 10 machines-10GB SFP+ dedicated network

### DISCUSSION THREAD

**Discussion by norburybrook** [apohl79/audiogridder#346] [Help]
I have a Cubase project i use for Audiogridder testing. it's a smallish project with a couple of gridder VI's and 10 or so AA aqua plugins and some Aether/B2 FX. I've noticed that my performance meter in cubase running this project is very high despite the fact there are NO plugins/FX/VI's at all on the host machine it's all coming from the slave machine. While everything runs fine it kind of defeats the object of having a slave/master system as I could run it all on the one machine with the same performance.

Any thoughts on this?

Main DAW:Ryzen 3950x  , 32 gig Ram-windows 10 10GB SFP+ network
Slave i7-3930K -32gig Ram-windows 10 -10GB SFP+ network

as you can see I have 2 great machines, a dedicated 10GB network for audio gridder so I don't think I could find any bottle necks here :)

cheers

m

**Comment by CaBuffalo** [comment]
It sounds like there are plugin(s) causing the heavy system loads. Take a look at the task manager of your slave system.How much CPU % it reaches? Or take a look at the log file under C:\Users\<User>\AppData\Roaming\AudioGridder\. I bet there are plugins giving you a lot of warning messages.

If it doesn't help you to disgonistic, send the dump file to Andreas. He will be able to take a look. 
https://audiogridder.com/submitting-large-files/

**Comment by apohl79** [comment]
Ok just to confirm my understanding
- the AG plugins have no instruments nor fx remote plugins loaded
- the VI/AA etc plugins are loaded directly on the cubase side

Now some questions:
- If you bypass the AG plugins, does the load meter go down?
- What buffer size are you using in AG? If the load meter is up, are you at 0 blocks? 
- If you increase the buffer size in AG, does the load go down? (thats what would be expected)

**Reply by norburybrook** [reply]
Andreas,

EVERYTHING in the project is running via Audiogridder on the slave machine via the plugin on the host machine. Yes, as I was playing VI's I had the buffer at zero. Increasing the buffer does help but even on maximum I'm still getting the Cubase performance meter advancing heavily . Perhaps Audio gridder is loading a single core on my AMD 3950x host machine which will mean the Cubase ASIO meter will rise. 

ideally I would like to be able to use the slave machine to STOP CPU /ASIO usage on the Main DAW and distribute the load better. 
M

**Reply by apohl79** [reply]
Ok so if everything is remote, the meter in cubase should go down, once you bypass the remotes in AG right?

In the plugin mon (ag plugin settings menu -> plugin monitor -> show), you can see the real network & processing latency of the audio data going to the server and back per channel/ag instance.

I think cubase (like logic) is metering the processing time of the insert. the more time an insert needs, the higher it goes. Thats by why buffering helps. You can also check, if you see buffering warning in the plugin logfile ("read queue empty..."). If that happens, AG is waiting for data and will probably cause audio stutters etc.

> Perhaps Audio gridder is loading a single core 

AG uses threads (each plugin instance creates an audio streaming thread for example) and does not try to bind to particular cores. But the audio processing happens in threads the DAW spawns.

> ideally I would like to be able to use the slave machine to STOP CPU /ASIO usage on the Main DAW and distribute the load better.

Thats exactly what I do. I run nothing locally. Every single channel goes via AG to a remote server.

**Reply by norburybrook** [reply]
thanks Andreas. Yes when I bypass the remote's the perf meter goes down in cubase. When I up the buffers to maximum the cpu goes down too. I have VE pro on these 2 machines and VS pro seems to add no additional overhead to the Cubase Perf meter regardless of how many VI's /reverbs I have loaded on the Slave. It will start to get drop out's when I'm maxing the slave CPU but I can up the buffers in my cubase project and that helps. Obviously gridder is different in how it works and is a lot easier for plugins etc. 

I was just wondering what I can do to optimise even more so I can max out the slave machine withoutj hitting the roof on the master machine. At the moment with my test project I'm not gaining anything as the perf meter isn't that different using everything in audio gridder from the slave than if I loaded the same things locally.

M

**Reply by apohl79** [reply]
Yeah I see. Unless you see these buffer warnings in the log, AG should mostly not wait for data and should add no load to the meter - in theory at least. :-)

But AG would wait if data is not available, to make sure that there is no audio breakage at mixdown.

But performance can be improved for sure. There is couple if things i'm planning to change, for instance the streaming protocol. That's should help i guess.

Thanks for these kind of tests! 

Andreas

**Reply by norburybrook** [reply]
Andreas I checked the logs and there are quite a few waiting warnings for AA plugins. What's the best way to log a session and check it? Latency/buffer size on both the host and server makes a big difference to the perf meter in Cubase when using audiogridder, more than using local plugins.

**Reply by apohl79** [reply]
Yes and your network performance matters a lot as well. I think you need to try different settings to find your sweet spot. In my setup, the server is just a few meters away with a single switch in between. I can easily use 30+ mono channels with no buffering in AG and 2k samples buffer in cubase. I would say i have 3-5 inserts per channel. When starting to add more heavy mastering plugins like ozone to the mix bus, i usually have to go to 1 block buffering in AG.

If you see those waiting warnings, you might have network bottlenecks, that pop up from time to time, like if some other devices on your network use it or so. Hard to say.

**Comment by norburybrook** [comment]
Andreas, Thanks for the reply. My machines are next to each other and connected with a 1m cable. The network is a 10Gb/s dedicated audiogridder network only so there is no other traffic on there, so I don’t think the network is a bottle neck at all as I’m using 2x SFP+ 10Gb network cards in each machine. 2k buffer in Cubase is very large 😊 I Only ever use 1k if I’m working on a particularly heavy mix. Normaly my machine runs @128 samples all day long.   I think the AA plugins are an issue as they still haven’t released their server version which has been planned for a few years now so there’s obviously something stopping them releasing the slave/host system they promised. I may try using the zero latency versions as a test today.  Thanks for the correspondence and continued development of what is a great product. It’s funny, many many years ago I had a 3 computer system set up using a similar technique with software called FX teleport which worked remarkably well but then disappeared. I’ve often wondered what happened to the FX teleport dev, perhaps he went to work for Vienna to help them develop VE pro?   Cheers Marcus     Sent from Mail for Windows 10 From: Andreas PohlSent: 02 February 2021 10:01To: apohl79/audiogridderCc: norburybrook; AuthorSubject: Re: [apohl79/audiogridder] High Host CPU use. Cubase 11-2x windows 10 machines-10GB SFP+ dedicated network (#346) Yes and your network performance matters a lot as well. I think you need to try different settings to find your sweet spot. In my setup, the server is just a few meters away with a single switch in between. I can easily use 30+ mono channels with no buffering in AG and 2k samples buffer in cubase. I would say i have 3-5 inserts per channel. When starting to add more heavy mastering plugins like ozone to the mix bus, i usually have to go to 1 block buffering in AG.If you see those waiting warnings, you might have network bottlenecks, that pop up from time to time, like if some other devices on your network use it or so. Hard to say.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Reply by apohl79** [reply]
Ok with that setup, it should not be the network for sure.

Yeah 2k is large, but not relevant when mixing for me. :-)

And no idea about the AA plans or what happened to FX teleport. There is is still a lot to do on the AG side, so thx for using it, getting feedback etc!
